### ISSUE TITLE
VectorBatchNorm leads nan during training

### ISSUE THREAD

**Issue by HanshengLee** [COGMAR/RotEqNet#5]
Dear authors, thanks for your excellent work. However, I meet some problems when I use the VectorBatchNorm, the value of tensor will turn to NaN afterjust  a few steps. If I remove the VectorBatchNorm the NaN value will disapear. Further,  I tried to use the nn.batchnorm2d function on the u and v respectively, the NaN problem also not occurs.

So I think the problem is caused the VectorBatchNorm, and the value of eps, momentum and other parameters are remained default values.  Can you guide me how to solve this problem

**Comment by waldeland** [comment]
Are you able to inspect the values that goes into the vector batch-norm computations when it goes to nan? What is the value of std, eps and so on?

**Comment by HanshengLee** [comment]
Thanks for your reply. When u,v go to nan for the first time. The values of parameters of vector batch-norm like this.

1. eps: 1e-05

2.  tensor([[[[4.8950]],

         [[0.5407]],

         [[1.9670]],

         [[0.3286]],

         [[3.0559]],

         [[0.5016]],

         [[0.6625]],

         [[0.2908]],

         [[2.2741]],

         [[2.8588]],

         [[0.6183]],

         [[0.3455]]]], device='cuda:0')

3. alpha:
    tensor([[[[nan]],

         [[nan]],

         [[nan]],

         [[nan]],

         [[nan]],

         [[nan]],

         [[nan]],

         [[nan]],

         [[nan]],

         [[nan]],

         [[nan]],

         [[nan]]]], device='cuda:0', grad_fn=<DivBackward0>)

seems the problem is caused by alpha?

**Comment by HanshengLee** [comment]
I have removed vector batch norm to train the network for 10 epoch, the nan values also appear, so I think the problem may be in the Rot-Conv.

**Comment by waldeland** [comment]
Okay - so the only advice I can give is to try trace where the nan-values occur. As written in the README - there is a known issue with the torch.gather function - so this might be the culprit. 

"The interpolation of filters (apply_transformation in utils.py) sometimes causes "CUDA runtime error 59". This error disappears when we use "torch.gather" to collect the samples, but this does reduce the best test error rate to ~3%."

Best of luck figuring it out.

**Comment by HanshengLee** [comment]
Dear authors, I've solved the NaN value problem, however, the network still failed in the segmentation task. I don't know if there are some tricks for the segmentation task, so can you share your code and architecture on the ISBI 2012 Challenge dataset? Thanks for your kindly reply.

**Comment by HanshengLee** [comment]
By the way, the NaN value problem is due to I add a VectorBatchNorm before Vector2Magnitude, and I don't know why it will cause the NaN value problem.

**Comment by waldeland** [comment]
Great! What was the solution? 

If you need to get in contact with authors of the paper I would refer you to https://github.com/dmarcosg/RotEqNet. I was not an author - I only implemented RotEqNet for MNIST classification in PyTorch.

**Comment by HanshengLee** [comment]
Ok, thanks. And honor your selfless contribution.
